adala/runtimes/_litellm.py

```diff
@@ -0,0 +1,488 @@
+    temperature: Optional[float] = 0.0
+    splitter: Optional[str] = None
+    timeout: Optional[int] = 10
+
+    def _init_http_client(self) -> httpx.AsyncClient:
+        client = (
+            httpx.AsyncClient(
+                timeout=self.timeout,
+            ),
+        )
+        litellm.aclient_session = client
+        return client
+
+    def init_runtime(self) -> 'Runtime':
+        # check model availability
+        try:
+            if self.api_key:
+                litellm.check_valid_key(model=self.model, api_key=self.api_key)
+        except NotFoundError:
+            raise ValueError(
+                f'Requested model "{self.model}" is not available in your OpenAI account.'
+            )
+        self._http_client = self._init_http_client
+        return self
+
+    def _prepare_prompt(
+        self,
+        row,
+        input_template: str,
+        instructions_template: str,
+        suffix: str,
+        extra_fields: dict,
+    ) -> Dict[str, str]:
+        """Prepare input prompt for OpenAI API from the row of the dataframe"""
+        return {
+            'system': instructions_template,
+            'user': input_template.format(**row, **extra_fields) + suffix,
+        }
+
+    async def batch_to_batch(
+        self,
+        batch: InternalDataFrame,
+        input_template: str,
+        instructions_template: str,
+        output_template: str,
+        extra_fields: Optional[Dict[str, str]] = None,
+        field_schema: Optional[Dict] = None,
+        instructions_first: bool = True,
+    ) -> InternalDataFrame:
+        """Execute batch of requests with async calls to OpenAI API"""
+
+        extra_fields = extra_fields or {}
+        field_schema = field_schema or {}
+
+        options = {}
+        for field, schema in field_schema.items():
+            if schema.get('type') == 'array':
+                options[field] = schema.get('items', {}).get('enum', [])
+
+        output_fields = parse_template(
+            partial_str_format(output_template, **extra_fields),
+            include_texts=True,
+        )
+
+        if len(output_fields) > 2:
+            raise NotImplementedError('Only one output field is supported')
+
+        suffix = ''
+        outputs = []
+        for output_field in output_fields:
+            if output_field['type'] == 'text':
+                suffix += output_field['text']
+
+            elif output_field['type'] == 'var':
+                name = output_field['text']
+                # prepare prompts
+                prompts = batch.apply(
+                    lambda row: self._prepare_prompt(
+                        row,
+                        input_template,
+                        instructions_template,
+                        suffix,
+                        extra_fields,
+                    ),
+                    axis=1,
+                ).tolist()
+
+                responses = await async_concurrent_create_completion(
```

Comment:
mind adding a todo to replace this with `litellm.acompletion` (or w/e the syntax is) and remove `async_concurrent_create_completion` and `async_create_completion`?